incubator/hnc/internal/validators/object.go

```diff
@@ -112,22 +112,24 @@ func (o *Object) handle(ctx context.Context, log logr.Logger, op admissionv1beta
 	// source can be created without causing any conflict.
 	if !oldInherited && !newInherited {
 		// check selector format
-		err := validateSelectorChange(inst, oldInst)
 		// If this is a selector change, and the new selector is not valid, we'll deny this operation
-		if err != nil {
+		if err := validateSelectorChange(inst, oldInst); err != nil {
 			msg := fmt.Sprintf("Invalid Kubernetes labelSelector: %s", err)
 			return deny(metav1.StatusReasonBadRequest, msg)
 		}
-		err = validateTreeSelectorChange(inst, oldInst)
-		if err != nil {
+		if err := validateTreeSelectorChange(inst, oldInst); err != nil {
 			msg := fmt.Sprintf("Invalid HNC %q value: %s", api.AnnotationTreeSelector, err)
 			return deny(metav1.StatusReasonBadRequest, msg)
 		}
-		err = validateNoneSelectorChange(inst, oldInst)
-		if err != nil {
+		if err := validateNoneSelectorChange(inst, oldInst); err != nil {
 			return deny(metav1.StatusReasonBadRequest, err.Error())
 		}
-		// TODO(@ginnyji): modify hasConflict so that it's aware of selectors
+		if err := validateSelectorUniqueness(inst, oldInst); err != nil {
+			return deny(metav1.StatusReasonBadRequest, err.Error()+
+				", only one selector annotation should exist."+
+				" Please delete the existing one before adding a new one.")
```

Comment:
There might not be an "existing one" - this might just be a yaml file with two selectors. This will changes how you implement `validateSelectorUniqueness`.

Also, I'd just return the full string from `validateSelectorUniqueness` and display that, e.g.:

`if msg := validateSelectorUniqueness(...); msg != "" {`

There's no need to convert from a string to an error and back again.

Reply:
Just confirming the `validateSelectorUniqueness` implementation actually did implement the case for a new yaml file with two selectors. So do the tests, right?

Reply:
Yeah the test case currently covers the case where we creating multiple selectors at the same time. I modified the error msg to accommodate this case better

Reply:
sg